python/ml4ir/applications/classification/pipeline.py

```diff
@@ -164,6 +166,102 @@ def get_relevance_dataset(
+            `pd.DataFrame` containing groupwise metrics if
+            group_metric_keys are defined in the FeatureConfig
+        metrics_dict : dict
+            metrics as a dictionary of metric names mapping to values
+
+        Notes
+        -----
+        You can directly do a `model.evaluate()` only if the keras model is compiled
+        """
+        relevance_model = self.get_relevance_model()
+        if not relevance_model.is_compiled:
+            return NotImplementedError
+
+        group_metrics_keys = self.feature_config.get_group_metrics_keys()
+
+        metrics_dict = relevance_model.model.evaluate(test_dataset)
+        metrics_dict = dict(zip(relevance_model.model.metrics_names, metrics_dict))
+
+        predictions = relevance_model.predict(test_dataset,
+                                              inference_signature=inference_signature,
+                                              additional_features=additional_features,
+                                              logs_dir=None,  # Return pd.DataFrame of predictions
+                                              logging_frequency=logging_frequency)
+        # Need to convert predictions from tuples of tf.Tensor to lists of int/float
+        # so that we are compatible with tf.keras.metrics
+        label_name = self.feature_config.get_label()['name']
+        output_name = relevance_model.output_name
+        predictions[label_name] = predictions[label_name].apply(lambda l: [item.numpy() for item in l])
+        predictions[output_name] = predictions[output_name].apply(lambda l: [item.numpy() for item in l])
+
+        global_metrics = []  # group_name, metric, value
+        grouped_metrics = []
+        for metric in relevance_model.model.metrics:
+            metric.reset_states()
+            metric.update_state(predictions[label_name].values.tolist(), predictions[output_name].values.tolist())
+            global_metrics.append({"metric": metric.name,
+                                   "value": metric.result().numpy()})
+            for group_ in group_metrics_keys:
+                for name, group in predictions.groupby(group_['name']):
+                    if group.shape[0] >= group_metrics_min_queries:
+                        metric.reset_states()
+                        metric.update_state(group[label_name].values.tolist(),
+                                            group[output_name].values.tolist())
+                        grouped_metrics.append({"group_name": group_['name'],
+                                                "group_key": name,
+                                                "metric": metric.name,
+                                                "value": metric.result().numpy(),
+                                                "size": group.shape[0]})
+        global_metrics = pd.DataFrame(global_metrics)
+        grouped_metrics = pd.DataFrame(grouped_metrics)
+        if logs_dir:
```

Comment:
results (`global_metrics` and `group_metrics`) are written to the logs_dir as CSV fles.
cc: @Ullimague